platform/web/request.go

```diff
@@ -147,6 +153,9 @@ func decode(r *http.Request, val interface{}, options DecoderOptions) error {
 	}
 
 	if err := decoder.Decode(val); err != nil {
+		if err.Error() == "http: request body too large" {
+			return fmt.Errorf("%w", NewErrRequestBodyTooLarge())
+		}
```

Comment:
I realized that we sometimes use `Decode(r *http.Request, val interface{}) error` directly and it changes the place where we first read the body, hence why we require to catch the error here as well.

It bugs me a bit that when we use `DecodeWithEmbeddedJSONSchema()` we actually end up remaking a buffer for the request body to pass it to `decode()` internally which fills the Go struct with the JSON data.

But apart from changing 
```go
func decode(r *http.Request, val interface{}, options DecoderOptions) error 
```
 to take a `[]byte` for the request body instead of the full request and its `io.Reader` for the body and use `json.Unmarshal` and make sure we read the initial request body buffer once I don't really see a better solution.
 
 But then we lose the `DisallowUnknownFields` from `json.Decoder` which is what we wanted 🤔 
 
@julien-leclercq Does it seems strange to you as well, or is it just me 😅  ?